### PR TITLE
Add Fusion Viewer tab to Patient View

### PR DESCRIPTION
**Pull request by johnoooh**
## Summary

- Adds a new **Fusion Viewer** tab to the cBioPortal patient view that visualizes gene fusions from structural variant data
- Converts cBioPortal `StructuralVariant` records into a unified `FusionEvent` model and fetches transcript/exon data from Genome Nexus
- Renders an interactive SVG diagram showing gene tracks, exon structure, fusion product, protein domains, and connecting arcs at the breakpoint junction
- Tab is conditionally shown only when structural variant data is available for the patient

image

[Example preview case](https://deploy-preview-5483.preview.cbioportal.org/patient/fusionViewer?studyId=msk_chord_2024&caseId=P-0002894)

## Motivation

The MSK-TARGET panel generates gene fusion calls and we wanted a way to visualize them in the portal and give investigators the ability to compare transcripts.  The portal also has a large amount of tabular structural variant data. 

Ritika mentioned that the fusion files are easily represented by the current tabular structural variant data.  We convert Fusion data as well as StructuralVariant data to a FusionEvent, which is then visualized in the Fusion tab

This tab provides a purpose-built visualization that shows the genomic context of each fusion — which exons are retained, where the breakpoint falls, and what protein domains are affected — making it easier for reviewers to interpret fusion events during case review.

## What's included

**Data layer** (4 files):
- `structuralVariantAdapter.ts` — converts `StructuralVariant[]` → `FusionEvent[]` with strand inference, read support computation, and intragenic rearrangement detection
- `genomeNexusService.ts` — fetches transcript/exon/PFAM domain data from Genome Nexus with caching, canonical transcript tagging, and GRCh37/38 build support
- `types.ts` — shared interfaces (`FusionEvent`, `GenePartner`, `TranscriptData`, `Exon`, `ProteinDomain`)
- `fusionProductHelpers.ts` — pure functions for 5'/3' exon selection and junction position computation

**Store** (1 file):
- `FusionViewerStore.ts` — MobX store managing fusion selection, multi-transcript selection, async transcript fetching with race condition guards, and genome build switching

**Components** (9 files):
- `FusionViewerTab.tsx` — top-level tab component with observer pattern
- `FusionListSidebar.tsx` — selectable fusion list with read support and frame info
- `FusionInfoBar.tsx` — header showing gene symbols, position, and transcript selector dropdowns
- `FusionDiagramSVG.tsx` — SVG orchestrator composing gene tracks, fusion product, and arcs
- `GeneTrack.tsx` — renders gene exon structure with breakpoint marker
- `FusionProduct.tsx` — renders the chimeric fusion product with retained exons
- `ConnectingArcs.tsx` — draws bezier curves connecting gene tracks to fusion product
- `ProteinDomainTrack.tsx` — displays PFAM protein domains below gene tracks
- `ExonTooltip.tsx` — hover tooltip showing exon number, coordinates, and length

**Tab registration** (1 file):
- `PatientViewPageTabs.tsx` — registers the Fusion Viewer tab with conditional visibility based on structural variant data availability

**Tests** (4 files, 90 tests):
- `structuralVariantAdapter.spec.ts` — field mapping, strand inference, read support, NA handling, intragenic detection, filtering
- `ensemblTranscriptService.spec.ts` — transcript mapping, canonical tagging, PFAM resolution, isForteSelected priority chain, cache clone protection
- `FusionViewerStore.spec.ts` — store actions, computed getters, race condition guard, toggle constraints, genome build switching
- `fusionProductHelpers.spec.ts` — exon selection for ±strands, boundary conditions, junction position math

## Test plan

- [ ] Verify tab appears only when patient has structural variant data
- [ ] Select different fusions from the sidebar and confirm diagram updates
- [ ] Switch between GRCh38/GRCh37 builds and verify transcript data reloads
- [ ] Hover over exons to confirm tooltip shows correct coordinates
- [ ] Verify multi-transcript selection checkboxes work (cannot deselect last)
- [ ] Check that single-gene fusions (no gene2) render without errors
- [ ] Run `yarn run testMain` — all 90 new tests pass

### CODE REVIEW

File: src/pages/patientView/fusionViewer/FusionViewerStore.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    public selectFusion(fusionId: string): void {
+        this.selectedFusionId = fusionId;
+
+        // Reset transcript selections
+        this.selectedTranscript5pIds.clear();
+        this.selectedTranscript3pIds.clear();
+
+        const fusion = this.fusions.find(f => f.id === fusionId);
+        if (!fusion) return;
+
```
[the remark this answers is not in the document]
Comment (johnoooh) [reply src/pages/patientView/fusionViewer/FusionViewerStore.ts]: Resolved by claude in commit 8c60c91

File: src/pages/patientView/fusionViewer/FusionListSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                                    if (!selected) {
+                                        (e.currentTarget as HTMLElement).style.backgroundColor =
+                                            '#f0f0f0';
+                                    }
+                                }}
+                                onMouseLeave={e => {
+                                    (e.currentTarget as HTMLElement).style.backgroundColor = selected
+                                        ? '#e8eef7'
+                                        : '#fff';
+                                }}
```
[the remark this answers is not in the document]
Comment (johnoooh) [reply src/pages/patientView/fusionViewer/FusionListSidebar.tsx]: Resolved in commit 8c60c91

File: src/pages/patientView/fusionViewer/FusionViewerStore.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                const forte = result.find(
+                    (t: TranscriptData) => t.isForteSelected
+                );
+                if (forte) {
+                    this.selectedTranscript3pIds.add(forte.transcriptId);
+                } else if (result.length > 0) {
+                    this.selectedTranscript3pIds.add(result[0].transcriptId);
+                }
+            }
+        },
```
[the remark this answers is not in the document]
Comment (johnoooh) [reply src/pages/patientView/fusionViewer/FusionViewerStore.ts]: Resolved in 8c60c91

File: src/pages/patientView/fusionViewer/FusionViewerStore.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            if (this.selectedTranscript5pIds.size === 0 && result) {
+                const forte = result.find(
+                    (t: TranscriptData) => t.isForteSelected
+                );
+                if (forte) {
+                    this.selectedTranscript5pIds.add(forte.transcriptId);
+                } else if (result.length > 0) {
+                    this.selectedTranscript5pIds.add(result[0].transcriptId);
+                }
+            }
```
[the remark this answers is not in the document]
Comment (johnoooh) [reply src/pages/patientView/fusionViewer/FusionViewerStore.ts]: Resolved in 8c60c91

File: src/pages/patientView/fusionViewer/FusionViewerTab.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+@observer
+export class FusionViewerTab extends React.Component<IFusionViewerTabProps> {
+    private store: FusionViewerStore;
+
+    constructor(props: IFusionViewerTabProps) {
+        super(props);
+        this.store = new FusionViewerStore();
+        this.store.setStructuralVariants(props.structuralVariants);
+    }
+
```
[the remark this answers is not in the document]
Comment (johnoooh) [reply src/pages/patientView/fusionViewer/FusionViewerTab.tsx:36]: Resolved in 8c60c91
